File: CS/SchedulerMappingConverterExample/MappingConverterLabel.cs
using DevExpress.XtraScheduler;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.Collections;

namespace SchedulerMappingConverterExample {
    #region #MappingConverterLabel
    class MappingConverterLabel : ISchedulerMappingConverter {
        SchedulerStorage storage;

        public MappingConverterLabel(SchedulerStorage dataStorage) {
            storage = dataStorage;
        }

        public object Convert(object obj, Type targetType, object parameter) {
            IAppointmentLabel lbl = storage.Appointments.Labels.FirstOrDefault(label => label.GetColor().ToArgb() == (int)obj);
            return (lbl != null) ? lbl.Id : CreateNewLabel((int)obj);
        }

        public object ConvertBack(object obj, Type targetType, object parameter) {
            return storage.GetLabelColor(obj).ToArgb();
        }

        private object CreateNewLabel(int intColor) {
            Color color = Color.FromArgb(intColor);
            string labelName = "x" + intColor.ToString("X");
            foreach (KnownColor kColor in Enum.GetValues(typeof(KnownColor))) {
                Color known = Color.FromKnownColor(kColor);
                if (intColor == known.ToArgb()) {
                    labelName = known.Name;
                }
            }
            AppointmentLabel label = storage.Appointments.Labels.CreateNewLabel(labelName);
            label.SetColor(color);
            storage.Appointments.Labels.Add(label);
            return label.Id;
        }
    }
    #endregion #MappingConverterLabel
}

File: VB/SchedulerMappingConverterExample/MappingConverterLabel.vb
Imports DevExpress.XtraScheduler
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Drawing
Imports System.Collections

Namespace SchedulerMappingConverterExample
    #Region "#MappingConverterLabel"
    Friend Class MappingConverterLabel
        Implements ISchedulerMappingConverter

        Private storage As SchedulerStorage

        Public Sub New(ByVal dataStorage As SchedulerStorage)
            storage = dataStorage
        End Sub

        Public Function Convert(ByVal obj As Object, ByVal targetType As Type, ByVal parameter As Object) As Object Implements ISchedulerMappingConverter.Convert
            Dim lbl As IAppointmentLabel = storage.Appointments.Labels.FirstOrDefault(Function(label) label.GetColor().ToArgb() = DirectCast(obj, Integer))
            Return If(lbl IsNot Nothing, lbl.Id, CreateNewLabel(DirectCast(obj, Integer)))
        End Function

        Public Function ConvertBack(ByVal obj As Object, ByVal targetType As Type, ByVal parameter As Object) As Object Implements ISchedulerMappingConverter.ConvertBack
            Return storage.GetLabelColor(obj).ToArgb()
        End Function

        Private Function CreateNewLabel(ByVal intColor As Integer) As Object
            Dim color As Color = System.Drawing.Color.FromArgb(intColor)
            Dim labelName As String = "x" & intColor.ToString("X")
            For Each kColor As KnownColor In System.Enum.GetValues(GetType(KnownColor))
                Dim known As Color = System.Drawing.Color.FromKnownColor(kColor)
                If intColor = known.ToArgb() Then
                    labelName = known.Name
                End If
            Next kColor
            Dim label As AppointmentLabel = storage.Appointments.Labels.CreateNewLabel(labelName)
            label.SetColor(color)
            storage.Appointments.Labels.Add(label)
            Return label.Id
        End Function
    End Class
    #End Region ' #MappingConverterLabel
End Namespace
